src/wsi/wsi_window.h

```diff
@@ -0,0 +1,125 @@
+#pragma once
+
+// The windows.h will define HWND/HMONITOR
+// to be types that it wants for the wsi we are compiling against.
+// ie. HWND     = their own window data ptr (eg. SDL_Window)
+//     HMONTIOR = their own monitor data ptr/a display index
```

Comment:
This is **really** ugly.

I think it's better to have something like `wsi/win32/wsi_win32.h` where we include the real `windows.h`, and for other WSI implementations we could have `wsi/sdl2/wsi_sdl2.h` do something like
```c++
// this is kinda fine since both are pointer types
using HWND = SDL_Window*;

// this is more annoying since SDL doesn't have a display type and 0 is a valid display index
struct WsiSdl2Monitor;
using HMONITOR = WsiSdl2Monitor*;

inline HMONITOR toMonitorHandle(int index) {
  uintptr_t handleValue = index < 0 ? uintptr_t(0) : uintptr_t(index) + 1;
  return reinterpret_cast<HMONITOR>(index);
}

inline int fromMonitorHandle(HMONITOR handle) {
  auto handleValue = reinterpret_cast<uintptr_t>(handle);
  return (handleValue && handleValue != INVALID_HANDLE_VALUE)
    ? int32_t(handleValue - 1)
    : -1;
}
```

And then have a header `wsi/wsi.h` or something that includes the platform-specific header.

If we need more Win32 types (we probably do), those should be defined in another common header that we include on non-win32, and that is preferably **not** called `windows.h`.

Reply:
Can we at least get rid of the comment then to make it less obvious how utterly cursed and broken the whole thing is?

Reply:
Yes, it is also not true anymore actually since this got refactored. HWND/HMONITOR Is always a HANDLE so size is consistent with Windows.

The WSI headers define helper functions to cast to/from.

Reply:
That's okay then.